### PR TITLE
fix logic in npmInstallWithCheck

### PR DESCRIPTION
**Pull request by AlCalzone**
Found while investigating https://github.com/ioBroker/ioBroker.js-controller/issues/1782

This made it look like there is an error detecting the npm version when in fact there is an error installing.

### CODE REVIEW

**Review by foxriver76 — APPROVED** [review]

File: packages/controller/lib/setup/setupInstall.js [hunk 1]
```diff
@@ -289,7 +289,11 @@ function Install(options) {
             console.error('Assuming that correct version is installed.');
         }
 
-        return this._npmInstall(npmUrl, options, debug);
+        try {
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupInstall.js:292]: catch from outside? return await is a bit strange imo ;-)

Comment (foxriver76) [reply packages/controller/lib/setup/setupInstall.js:292]: but yes was this not so nice style before

Comment (AlCalzone) [reply packages/controller/lib/setup/setupInstall.js:292]: I think catching is the only reason to use return await. Catching from the outside means we have to catch and log in multiple locations though.

Comment (Apollon77) [reply packages/controller/lib/setup/setupInstall.js:292]: I would also leave like this for now

**Pull request merged**